cmd/swarm/main.go

```diff
@@ -110,16 +105,11 @@ var (
 		Usage:  "Swarm Syncing enabled (default true)",
 		EnvVar: SWARM_ENV_SYNC_ENABLE,
 	}
-	EnsAPIFlag = cli.StringFlag{
+	EnsAPIFlag = cli.StringSliceFlag{
 		Name:   "ens-api",
-		Usage:  "URL of the Ethereum API provider to use for ENS record lookups",
+		Usage:  "ENS API endpoint for a TLD and with contract address, can be repeated, format [tld:][contract-addr@]url",
```

Comment:
Have you tried to see if it works when the option is provided via config file?

Reply:
What is the behavior if only addresses with TLDs are provided, do we end up without a default one? Is it even possible to identify the default one if no default has been provided?

Reply:
The toml configuration file is loading the values this way:

EnsAPIs = [
        "endpoint1",
        "test:endpoint2",
]

The ENS API endpoint configuration is implemented to be most explicit. If no default endpoint is provided, no assumption are made. Could there be a case that no default resolver is required?

Reply:
I am concerned about the other way around; let's say user provides `.eth:0x....@geth.ipc`, what happens to `.test` and other TLDs?

Reply:
My view is that in that case, other domains except under .eth TLD should not be resolved, as that is configured by the user. In this particular case, one may assume that provided endpoint which is ti only one can be used for other TLDs. But what is the case if endpoints for two or more TLDs are provided, but not a default endpoint? One option is not to have the default resolver, and the order one is to add all resolvers as default resolvers. Maybe the later behaviour is a better solution?

Reply:
I am not sure I understand, are you saying that if explicit TLD resolvers are provided but no default resolver, that all are tried as a default resolver? I think that would probably be the best option. Let's have a second opinion. @lmars ? @homotopycolimit

Reply:
Yes, that is right. I would also like to hear more opinions.

Reply:
I think if the user doesn't provide a default (i.e. a value without the `tld:` part), then names which don't match an endpoint should not be resolved.

For example, if I explicitly start swarm with:

```
--ens-api .eth:<addr>@geth.ipc --ens-api .test:<addr>@testnet/geth.ipc
```

I would not expect say `.foo` requests to be resolved, rather an error be returned like `no ENS endpoint configured to resolve .foo names`.

Reply:
@lmars the functionality is now similar, just the error returned from MultiResolver.Resolve should include the information about the TLD, currently it is just an error (var errNoResolver) with "no resolver" message.

Reply:
ok, so @janos when you have changed the error message please ping, then I can mark this as approved from my side. Thanks.

Reply:
There is a new error type NoResolverError for more verbose error message.